Dev-2/Dev-2/ChangeStringTransliteration.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            foreach (char i in str)
+            {
+                if (char.IsLetter(i) && (!Enumerable.Range(97, 122).Contains(i)))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+        /// <summary>
+        /// Autodect which alphabet string contain 
+        /// change transliteraton of string <param>
+        /// </summary>
+        /// <param name="str"></param>
+        /// <returns>return transliterated string</returns>
+        public String ChangeTransliteration(String str)
+        {
+            String temporaryString = String.Empty;
+            temporaryString = str;
+            try
+            {
+                StringToLowerCase(ref temporaryString);
+
+                if (IsCyrillic(temporaryString))
+                {
+                    CyrillicToTransliteration(ref temporaryString);
+                }
+                else if (IsLatin(temporaryString))
+                {
+                    LatinToTransliteration(ref temporaryString);
+                }
+                else
+                {
+                    Console.WriteLine("The initial sting conains russian and english symbols");
+                }
+                return temporaryString;
+            }
+            catch 
+            {
+                throw new Exception("Can't translit this tring");
+            }
+        }
+        /// <summary>
+        /// replace each element in string ,using the dictionary
+        /// </summary>
+        /// <param name="str">reference on string</param>
+        private  void CyrillicToTransliteration (ref String str)
+        {
+            StringBuilder temporaryString = new StringBuilder();
+            temporaryString.Append(str);
+
+            Dictionary<String, String> cyrillicTransliterations = new Dictionary<String, String>
+            {
+                ["а"] = "a",
+                ["б"] = "b",
+                ["в"] = "v",
+                ["г"] = "g",
+                ["д"] = "d",
+                ["е"] = "e",
+                ["ё"] = "yo",
+                ["ж"] = "zh",
+                ["з"] = "z",
+                ["и"] = "i",
+                ["й"] = "y",
+                ["к"] = "k",
+                ["л"] = "l",
+                ["м"] = "m",
+                ["н"] = "n",
+                ["о"] = "o",
+                ["п"] = "p",
+                ["р"] = "r",
+                ["с"] = "s",
+                ["т"] = "t",
+                ["у"] = "u",
+                ["ф"] = "f",
+                ["х"] = "kh",
+                ["ц"] = "ts",
+                ["ч"] = "ch",
+                ["ш"] = "sh",
+                ["щ"] = "sch",
+                ["ъ"] = String.Empty,
+                ["ы"] = "y",
+                ["ь"] = String.Empty,
+                ["э"] = "e",
+                ["ю"] = "yu",
+                ["я"] = "ya"
+            };
+
+            foreach (string i in cyrillicTransliterations.Keys)
```

Comment:
do not replace the whole string, in this case you process the whole string many times, not only once. Better to replace symbol by symbol